compose/service.py

```diff
@@ -1695,3 +1698,127 @@ def rewrite_build_path(path):
+            - memory (int): set memory limit for build
+            - memswap (int): Total memory (memory + swap), -1 to disable
+                swap
+            - cpushares (int): CPU shares (relative weight)
+            - cpusetcpus (str): CPUs in which to allow execution, e.g.,
+                ``"0-3"``, ``"0,1"``
+        decode (bool): If set to ``True``, the returned stream will be
+            decoded into dicts on the fly. Default ``False``
+        shmsize (int): Size of `/dev/shm` in bytes. The size must be
+            greater than 0. If omitted the system uses 64MB
+        labels (dict): A dictionary of labels to set on the image
+        cache_from (:py:class:`list`): A list of images used for build
+            cache resolution
+        target (str): Name of the build-stage to build in a multi-stage
+            Dockerfile
+        network_mode (str): networking mode for the run commands during
+            build
+        squash (bool): Squash the resulting images layers into a
+            single layer.
+        extra_hosts (dict): Extra hosts to add to /etc/hosts in building
+            containers, as a mapping of hostname to IP address.
+        platform (str): Platform in the format ``os[/arch[/variant]]``
+        isolation (str): Isolation technology used during build.
+            Default: `None`.
+    Returns:
+        A generator for the build output.
+    """
+    command_builder = _CommandBuilder()
+    command_builder.add_arg("--tag", tag)
+    command_builder.add_flag("--quiet", quiet)
+    _ = fileobj
+    command_builder.add_flag("--no-cache", nocache)
+    command_builder.add_flag("--rm", rm)
+    _ = timeout
+    _ = custom_context
+    _ = encoding
+    command_builder.add_flag("--pull", pull)
+    command_builder.add_flag("--force-rm", forcerm)
+
+    if dockerfile:
+        dockerfile = os.path.join(path, dockerfile)
+
+    command_builder.add_arg("--file", dockerfile)
+    command_builder.add_params("--build-arg", buildargs)
+    command_builder.add_arg("--memory", container_limits.get("memory"))
+
+    _ = decode
+    _ = gzip
+    command_builder.add_arg("--shm-size", shmsize)
+    command_builder.add_params("--label", labels)
+    command_builder.add_list("--cache-from", cache_from)
+    command_builder.add_arg("--target", target)
+    command_builder.add_arg("--network", network_mode)
+    command_builder.add_flag("--squash", squash)
+    command_builder.add_params("--add-host", extra_hosts)
+    command_builder.add_arg("--platform", platform)
+    command_builder.add_arg("--isolation", isolation)
+    command_builder.add_arg("--progress", "plain")
+    args = command_builder.build([path])
+
+    with subprocess.Popen(args, stdout=subprocess.PIPE, universal_newlines=True) as p:
+        while True:
+            line = p.stdout.readline()
+            if line == "":
+                break
+            line = line.replace("writing image sha256:", "Successfully built ")
```

Comment:
why?

Reply:
for bypass this.
https://github.com/docker/compose/blob/master/compose/service.py#L1110
```python
        for event in all_events:
            if 'stream' in event:
                match = re.search(r'Successfully built ([0-9a-f]+)', event.get('stream', ''))
                if match:
                    image_id = match.group(1)
```

Reply:
This doesn't seem robust.
Can we just append `Successfully built...` when it didn't appear from `docker build` CLI?

Reply:
I do not know how to get image_id from stdout on 'docker build --progress=plain'.
Do you have any idea?

Reply:
`--iidfile`?

Reply:
I tried to implement to use an iidfile method.